r2dbc-core/src/main/java/io/micronaut/r2dbc/health/R2dbcHealthQueryProvider.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright 2017-2021 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.r2dbc.health;
+
+import jakarta.inject.Singleton;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+
+/**
+ * Provides Health Query for selection version from database.
+ *
+ * @author Anton Kurako (GoodforGod)
+ * @since 2.0.1
+ */
+@Singleton
+public class R2dbcHealthQueryProvider {
+
+    /**
+     * Return only info about SQL server version and do not leak environment details.
+     * Example: 'Microsoft SQL Server 2017'
+     * @see <a href='https://www.mssqltips.com/sqlservertip/1140/how-to-tell-what-sql-server-version-you-are-running/'>MS SQL Version</a>
+     */
+    private static final String MSSQL_QUERY = "SELECT TOP 1 value FROM STRING_SPLIT(@@VERSION, '(');";
+    private static final String COMMON_QUERY = "SELECT version();";
+
+    private static final String POSTGRES = "PostgreSQL";
+    private static final String MARIADB = "MariaDB";
+    private static final String MYSQL = "MySQL";
+    private static final String MSSQL = "Microsoft SQL Server";
+
+    /**
+     * {@link io.r2dbc.spi.ConnectionFactoryMetadata#getName()} to SQL query for database version.
+     */
+    private final Map<String, String> metadataNameToQuery;
+
+    public R2dbcHealthQueryProvider() {
+        this.metadataNameToQuery = new HashMap<>(12);
```

Comment:
Please move this collection to `R2dbcHealthConfiguration` so it can be expanded/modified by configuration

Reply:
Just move it to R2dbcHealthConfiguration and inject it into Condition? Or you want this map also be filled via [@MapFormat](https://docs.micronaut.io/latest/api/io/micronaut/core/convert/format/MapFormat.html) ?

Reply:
Yes, map format might be needed to preserve corrct key names